app/page/channel.js

```diff
@@ -19,9 +23,25 @@ exports.create = function (api) {
     const { channel } = location
 
     const channelName = channel.substr(1)
+    const myKey = api.keys.sync.id()
+    var subscribed = api.channel.obs.subscribed(myKey).has(channelName)
+
+    function toggleSubscription () {
+      api.sbot.async.publish({
+        type: 'channel',
+        channel: channelName,
+        subscribed: !subscribed()
+      })
+    }
+
+    const subscribeButton = h('Button -subscribe',
+      { 'ev-click': toggleSubscription },
+      when(subscribed, 'Unsubscribe from channel', 'Subscribe to channel')
+    )
+
```

Comment:
Check out `toggleSubscription` and `subscribeButton` to see what I've changed.
Ask questions if there's anything that seems too magical here or unclear :)

p.s. I learnt more about mutant from your code here - it didn't know obs.subscribed.has was an obs as well !

Reply:
@arj03  ^^

Reply:
Yes makes sense. Still quite new to mutant, but this is clearly DRY :)